jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -835,6 +835,96 @@ ecma_builtin_helper_def_prop (ecma_object_t *obj_p, /**< object */
                                              &prop_desc);
 } /* ecma_builtin_helper_def_prop */
 
+#if ENABLED (JERRY_ES2015)
+
+/**
+ * The Object.prototype object's __proto__ accessor property getter
+ *
+ * See also:
+ *          ECMA-262 v6, B.2.2.1
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_helper_object_prototype_get_proto (ecma_value_t this_value) /**< this value */
+{
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_value);
+
+  if (obj_p->u2.prototype_cp == JMEM_CP_NULL)
+  {
+    return ECMA_VALUE_NULL;
+  }
+
+  ecma_object_t *proto_obj_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, obj_p->u2.prototype_cp);
+
+  ecma_ref_object (proto_obj_p);
+
+  return ecma_make_object_value (proto_obj_p);
+} /* ecma_builtin_helper_object_prototype_get_proto */
+
+/**
+ * The Object.prototype object's __proto__ accessor property setter
+ *
+ * See also:
+ *          ECMA-262 v6, B.2.2.1.1
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_helper_object_prototype_set_proto (ecma_value_t this_value, /**< this value */
+                                                ecma_value_t proto) /**< proto value */
+{
+  if (ECMA_IS_VALUE_ERROR (ecma_op_check_object_coercible (this_value)))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+
+  if ((!ecma_is_value_object (proto)
+      && !ecma_is_value_null (proto))
+      || !ecma_is_value_object (this_value))
+  {
+    return ECMA_VALUE_UNDEFINED;
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_value);
+
+  ecma_object_t *current_proto_p = ECMA_GET_POINTER (ecma_object_t, obj_p->u2.prototype_cp);
+  ecma_object_t *new_proto_p = ecma_is_value_null (proto) ? NULL : ecma_get_object_from_value (proto);
+
+  if (current_proto_p == new_proto_p)
+  {
+    return ECMA_VALUE_UNDEFINED;
+  }
+
+  if ((obj_p->type_flags_refs & ECMA_OBJECT_FLAG_EXTENSIBLE) == 0)
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Object is not extensibe!"));
+  }
+
+  ecma_object_t *iterator_p = new_proto_p;
+
+  while (true)
+  {
+    if (iterator_p == NULL)
+    {
+      break;
+    }
+
+    if (obj_p == iterator_p)
+    {
+      return ecma_raise_type_error (ECMA_ERR_MSG ("Cyclic __proto__ value"));
```

Comment:
We already have function for [Get/Set]PrototypeOf, can't we reuse those instead of duplicating the logic?